src/Utils/normalization.jl

```diff
@@ -69,31 +61,72 @@ function normalize(system::AbstractSystem)
     throw(ArgumentError("the system type $(typeof(system)) is currently not supported"))
 end
 
-# systems already in canonical form, i.e. which don't need normalization
-normalize(system::CF) = system
+# x' = Ax, in the continuous case
+# x+ = Ax, in the discrete case
+function normalize(system::LCS{N, AN}) where {N, AN<:AbstractMatrix{N}}
+    n = statedim(system)
+    X = Universe(n)
+    return CLCS(system.A, X)
+end
+
+function normalize(system::LDS{N, AN}) where {N, AN<:AbstractMatrix{N}}
+    n = statedim(system)
+    X = Universe(n)
+    return CLDS(system.A, X)
+end
+
+# x' = Ax, x ∈ X in the continuous case
+# x+ = Ax, x ∈ X in the discrete case
+for CL_S in (:CLCS, :CLDS)
+    @eval begin
+        function normalize(system::$CL_S{N, AN, ST}) where {N, AN<:AbstractMatrix{N}, ST<:LazySet{N}}
+            n = statedim(system)
+            X = _wrap_invariant(stateset(system), n)
+            return $CL_S(system.A, X)
+        end
+    end
+end
+
+# x' = Ax + u, x ∈ X, u ∈ U in the continuous case
+# x+ = Ax + u, x ∈ X, u ∈ U in the discrete case
+for CLC_X in (:CLCCS, :CLCDS)
+    @eval begin
+        function normalize(system::$CLC_X{N, IdentityMultiple{N}, ST}) where {N, ST<:LazySet{N}}
+            λ = system.B.M.λ
+            if λ == oneunit(λ)
+                return system
```

Comment:
You still need to wrap the inputs here (they could be a `LazySet`) and the invariant (note that here we still may want to allow for `nothing` because there is no suitable system type; but I am fine with requiring the user to specify a `Universe` instead).